redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/upgrade/action/RedbeamsUpgradeActions.java

```diff
@@ -0,0 +1,164 @@
+package com.sequenceiq.redbeams.flow.redbeams.upgrade.action;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.cloudbreak.cloud.scheduler.CancellationException;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.flow.core.Flow;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.redbeams.api.model.common.DetailedDBStackStatus;
+import com.sequenceiq.redbeams.domain.stack.DBStack;
+import com.sequenceiq.redbeams.flow.redbeams.common.RedbeamsContext;
+import com.sequenceiq.redbeams.flow.redbeams.common.RedbeamsEvent;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.AbstractRedbeamsUpgradeAction;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.RedbeamsUpgradeEvent;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.RedbeamsUpgradeState;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.event.BackupDatabaseServerRequest;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.event.BackupDatabaseServerSuccess;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.event.RedbeamsStartUpgradeRequest;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.event.RedbeamsUpgradeFailedEvent;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.event.RestoreDatabaseServerRequest;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.event.RestoreDatabaseServerSuccess;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.event.UpgradeDatabaseServerRequest;
+import com.sequenceiq.redbeams.flow.redbeams.upgrade.event.UpgradeDatabaseServerSuccess;
+import com.sequenceiq.redbeams.metrics.MetricType;
+import com.sequenceiq.redbeams.metrics.RedbeamsMetricService;
+import com.sequenceiq.redbeams.service.stack.DBStackStatusUpdater;
+
+@Configuration
+public class RedbeamsUpgradeActions {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RedbeamsUpgradeActions.class);
+
+    private static final String TARGET_MAJOR_VERSION = "TARGET_MAJOR_VERSION";
+
+    @Inject
+    private DBStackStatusUpdater dbStackStatusUpdater;
+
+    @Inject
+    private RedbeamsMetricService metricService;
+
+    @Bean(name = "BACKUP_DATABASE_SERVER_STATE")
+    public Action<?, ?> backupDatabaseServer() {
+        return new AbstractRedbeamsUpgradeAction<>(RedbeamsStartUpgradeRequest.class) {
+
+            @Override
+            protected void prepareExecution(RedbeamsStartUpgradeRequest payload, Map<Object, Object> variables) {
+                variables.put(TARGET_MAJOR_VERSION, payload.getMajorVersion());
```

Comment:
I would rather create a `RedbeamsUpgradeContext` and store `TARGET_MAJOR_VERSION` there, it seems to be more extensible in the future and can be used in each flow states

Reply:
I do not quite understand why putting major version to context is better than storing it in the variables. 
The context can get major_version from a payload, that is, the prerequisite to put it to context is its presence in a subclass of Payload. I will extend the request classes with the major version.